runtime/interpreter/value.go

```diff
@@ -17999,6 +17999,25 @@ func (v *CompositeValue) Iterator(interpreter *Interpreter, locationRange Locati
 	}
 }
 
+func (v *CompositeValue) ForEach(
+	interpreter *Interpreter,
+	_ sema.Type,
```

Comment:
What is this unused parameter for?

Reply:
This `sema.Type` parameter is the type of the value available for each iteration. This is only needed for the `ForEach` method of reference values since, for reference types, this is pre-calculated and needed to:
  1) Determine whether to return a reference or the concrete value as the iteration-value.
  2) Calculate the entitlements to be delegated to the iteration-value, if it is a reference.